### DISCUSSION TITLE
Umlaut characters not retrieved correctly

### DISCUSSION THREAD

**Discussion by minditer1** [dcm4che/dcm4che#1086] [Q&A]
### Discussed in https://github.com/dcm4che/dcm4che/discussions/1071

<div type='discussions-op-text'>

<sup>Originally posted by **minditer1** December  8, 2021</sup>
I am trying to use dcm4che (version 5.22.6) to query patient data from a specific PACS, using dcm4che as a maven dependency in a Spring Boot Java application. Everything is working well, except patient names that have german characters (umlauts). Instead of retrieving name that contains ö, dcm4che returns � instead. 

I did set UTF-8 character encoding in my Spring Boot application and backend does read and write successfully umlauts: ````<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>````

Tag specific character set is added on every request: 
````
public static void setCharacterSet(Attributes attributes) {
        int tag = Tag.SpecificCharacterSet;
        var vr = ElementDictionary.vrOf(tag, attributes.getPrivateCreator(tag));
        attributes.setString(tag, vr, "ISO_IR 100");
}
````
Is there anything else that I'm missing? 
</div>

**Comment by nroduit** [comment]
The encoding of the requests is done in dcm4che as described above but the encoding of the responses is done on the server-side and if the response does not contain any encoding then the default encoding is applied. See this [post](https://stackoverflow.com/questions/68497012/can-we-set-multiple-dcm-specificcharacterset-while-importing-records-using-dicom).

**Comment by minditer1** [comment]
> The encoding of the requests is done in dcm4che as described above but the encoding of the responses is done on the server-side and if the response does not contain any encoding then the default encoding is applied. See this [post](https://stackoverflow.com/questions/68497012/can-we-set-multiple-dcm-specificcharacterset-while-importing-records-using-dicom).

So if I understand correctly I am unable to control what character set is being used for returning the data?

**Reply by gunterze** [reply]
That's specified by DICOM - it's not specific to dcm4che!
(Typically DICOM archives will use the same Specific Character Set in responses, as in the DICOM Objects they received from sources.)

**Comment by nroduit** [comment]
Yes, that's right. dcm4che supports different encodings as long as it is passed in the answer. Have you checked if the encoding is specified in the response?
